### PR TITLE
Replace custom Result types on native

### PR DESCRIPTION
**Pull request by MortyMerr**
Swift 5.0 introduced Result type, so there is no need in custom implementations now

### CODE REVIEW

File: RxSwift/Traits/Single.swift [hunk 1]
```diff
@@ -15,13 +15,9 @@ public enum SingleTrait { }
...
-public enum SingleEvent<Element> {
-    /// One and only sequence element is produced. (underlying observable sequence emits: `.next(Element)`, `.completed`)
-    case success(Element)
-    
-    /// Sequence terminated with an error. (underlying observable sequence emits: `.error(Error)`)
-    case error(Swift.Error)
-}
+/// .success: One and only sequence element is produced. (underlying observable sequence emits: `.next(Element)`, `.completed`)
+/// .failure: Sequence terminated with an error. (underlying observable sequence emits: `.error(Error)`
+public typealias SingleEvent<Element> = Result<Element, Swift.Error>
```
Comment (freak4pc) [inline RxSwift/Traits/Single.swift]: The other changes make sense but I'm not sure this one necessarily does. 
Yes, the type signatures do - but they mean different things. Not too sure, I'll let @kzaher decide on this.

File: RxExample/RxExample/Result.swift [hunk 2]
```diff
@@ -1,12 +0,0 @@
-//
-//  Result.swift
-//  RxExample
-//
-//  Created by Krunoslav Zaher on 3/18/17.
-//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
-//
-
-enum Result<T, E: Error> {
```
Comment (freak4pc) [inline RxExample/RxExample/Result.swift]: This removal makes sense to me

**Comment by kzaher** [comment]
Hm ...,

I think that we could think about doing that in the future (6.0). We can't do anything at the moment because it would break ABI compatibility and semantic versioning.

We can only remove `RxExample/RxExample/Result.swift`.

**Comment by MortyMerr** [comment]
@kzaher ok thanks, i will separate that PR

File: RxExample/RxExample/Result.swift (hunk 2 above)
Comment (MortyMerr) [reply RxExample/RxExample/Result.swift]: Moved to separate PR

File: RxSwift/Traits/Single.swift (hunk 1 above)
Comment (MortyMerr) [reply RxSwift/Traits/Single.swift]: IMO, Result and SingleEvent have the same meaning here. I can't prove that but please just think about that a little bit more
Feel free to close that PR, if it's not suitable :)

**Pull request closed without merging**